internal/testrunner/test_config.go

```diff
@@ -15,11 +14,11 @@ type SkipConfig struct {
 	Reason string `config:"reason"`
 
 	// Link is a URL where more details about the skipped test can be found.
-	Link url.URL `config:"url"`
+	Link string `config:"link"`
```

Comment:
any other idea to try to keep `url.URL` here?

When the type is `url.URL` this is the error:
```
Error: error running package system tests: could not complete test run: unable to load system test case file '/home/user/Coding/work/integrations/packages/nats/data_stream/log/_dev/test/system/test-default-config.yml': unable to unpack system test configuration file: /home/user/Coding/work/integrations/packages/nats/data_stream/log/_dev/test/system/test-default-config.yml: required 'object', but found 'string' in field 'skip.link'
```

and this is the config.yml
```yaml
skip:
  reason: Test is skipped for now since it has been reported as flaky. We can rely on pipeline tests for now.
  link: https://github.com/elastic/integrations/issues/1576

```

Reply:
You could define a custom unmarshaler type, something like this:
```
type JSONURL struct {
        *url.URL
}

func (u *JSONURL) UnmarshalJSON(d []byte) error {
        var s string
        err := json.Unmarshal(d, &s)
        if err != nil {
                return err
        }

        u.URL, err = url.Parse(s)
        if err != nil {
                return err
        }

        return nil
}
```
```suggestion
	Link JSONURL `config:"link"`
```

Reply:
As this struct is read using `ucfg`, I've applied this suggestion using the Unpack method from this interface and it worked:
https://pkg.go.dev/github.com/elastic/go-ucfg#StringUnpacker

Thanks for the suggestion!!